bokehjs/src/coffee/core/properties.coffee

```diff
@@ -79,33 +79,27 @@ export class Property
 
     attr_value = obj.getv(attr)
 
-    if _.isUndefined(attr_value)
+    if attr_value == undefined
       default_value = @default_value
 
       attr_value = switch
-        when _.isUndefined(default_value) then null
-        when _.isArray(default_value)     then _.clone(default_value)
-        when _.isFunction(default_value)  then default_value(obj)
-        else                                   default_value
+        when default_value == undefined then null
+        when isArray(default_value)     then copy(default_value)
+        when isFunction(default_value)  then default_value(obj)
+        else                                 default_value
 
       obj.setv(attr, attr_value, {silent: true, defaults: true})
 
-    # if _.isObject(attr_value) and not _.isArray(attr_value) and not attr_value.properties?
-    #   @spec = attr_value
-    #   if _.size(_.pick.apply(null, [@spec].concat(@specifiers))) != 1
-    #     throw new Error("Invalid property specifier #{JSON.stringify(@spec)}, must have exactly one of #{@specifiers}")
-
-    if _.isArray(attr_value)
```

Comment:
The check is exactly as below (changed to more efficient version). I think the reason it's commented out is because it's too strict, e.g. doesn't take units into account. I think altogether this logic is reversed, because we already have type information and we are guessing the understanding of an object based on its content. I will start an issue for this, because this should be straightened out.